_test/test_sqw/test_noisify.m

```diff
@@ -0,0 +1,141 @@
+classdef test_noisify < TestCase
+
+properties
+    old_warn_state;
+
+    raw_pix_data = rand(9, 10);
+    small_page_size_ = 1e6;  % 1Mb
+    test_sqw_file_path = '../test_sqw_file/deterministic_sqw_fake_data_for_testing.sqw';
+    test_sqw_file_full_path = '';
+
+    pixel_data_obj;
+    pix_data_from_file;
+    pix_data_from_faccess;
+    pix_data_small_page;
+    pix_fields = {'u1', 'u2', 'u3', 'dE', 'coordinates', 'q_coordinates', ...
+                  'run_idx', 'detector_idx', 'energy_idx', 'signal', ...
+                  'variance'};
+end
+
+properties (Constant)
+    NUM_BYTES_IN_VALUE = 8;
+    NUM_COLS_IN_PIX_BLOCK = 9;
+end
+
+methods (Access = private)
+
+    function pix_data = get_random_pix_data_(~, rows)
+        data = rand(9, rows);
+        pix_data = PixelData(data);
+    end
+
+end
+
+methods
+
+    function obj = test_noisify(~)
+        obj = obj@TestCase('test_noisify');
+
+        % Swallow any warnings for when pixel page size set too small
+        obj.old_warn_state = warning('OFF', 'PIXELDATA:validate_mem_alloc');
+
+        % Process the file path
+        test_sqw_file = 'c:\boobly6.sqw';%java.io.File('c:\', obj.test_sqw_file_path);
+        obj.test_sqw_file_full_path = 'c:\boobly6.sqw';%char(test_sqw_file.getCanonicalPath());
+
+        % Tests use a number of different input streams:
+        % Construct an object from raw data
+        obj.pixel_data_obj = PixelData(obj.raw_pix_data);
+        % Construct an object from a file
+        obj.pix_data_from_file = PixelData(obj.test_sqw_file_path);
+        % Construct an object from a file accessor
+        f_accessor = sqw_formats_factory.instance().get_loader(obj.test_sqw_file_path);
+        obj.pix_data_from_faccess = PixelData(f_accessor);
+        % Construct an object from file accessor with small page size
+        obj.pix_data_small_page = PixelData(f_accessor, obj.small_page_size_);
+    end
+
+    function delete(obj)
+        warning(obj.old_warn_state);
+    end
+
+    % --- Tests for in-memory operations ---
+    
+    function test_how_noisify_works(obj)
+        % Thinking aloud here:
+        % obj is the test case object
+        % what we want to do is call noisify here on different kinds of
+        % data
+        % step 1 we reduce the page size
+        hc = hor_config();
+        hc.pixel_page_size = 10000;
+        % we set up the test "random number generator" which is actually
+        % a deterministic set of numbers 1:999 repeated. Use factor to make
+        % them in range 0:1
+        factor =1/999;
+        % We make an sqw object with the above page size
+        sqw_obj1 = sqw(obj.test_sqw_file_full_path);
+        % and we noisify it
+        rng(0) % don't need seed as it's not the proper rng but left in in
```

Comment:
We have a `seed_rng` function (in `_test/shared`) which sets a new random seed and returns the current state so it can be restored at the end of the test
```
[old_rng_state, rng_state] = seed_rng(fixed_seed);
clean_up_rng = onCleanup(@() rng(old_rng_state));
fprintf('RNG seed: %i\n', rng_state.Seed);
```
This was introduced to help stabilize the test execution of some tests that appear to be critically dependent on the RNG state.

If you're not using the system random number generator in this test it may be worth removing the `rng(0)` calls completely.

Reply:
In an ideal world I would like the opportunity for this test to be applied also for other random number distributions so that the test can be manually reused for a variety of scenarios and data. But accepting that this may be better done as a comment rather than live code.